### ISSUE TITLE
Jira integration

### ISSUE THREAD

**Issue by caioflores** [hangxingliu/vscode-coding-tracker#10]
Maybe the extension could integrate with Jira. An example, I could select in a list of Jira's tasks the task that I'm working on, than the time tracked while this could be linked to the task in graphs and could set the time spent in Jira card.

This feature could be very attractive, some IDEs have integration with Jira but there's no one good enough for Vscode (I didn't found).

**Comment by hangxingliu** [comment]
Sorry, I have never used Jira.  
Is integration you mentioned  means like this:

1. This extension provide a select panel which you can select tasks in your team Jira. And tasks you selected means you start work on.
2. This extension could detect your commit for these tasks, then calculate the time you elapsed in these tasks. And the time data will be sent to Jira (Looks like a Jira bot could report time cost on task pages).
3. And you can watch this data as a special chart in report page of this extension.

**Comment by caioflores** [comment]
Exactly. Jira has the option to log hours worked, this could be set automatically when we commit. 

One of Jetbrain's IDEs has something like that, I think that's ItelliJ. It's something like this:

**1 - Select a task in a list of tasks from Jira (from active sprint probably).**
When you select a task the IDE creates a new branch with the task name and checkouts in it.

**2 - Start the timer** 
In this case the IDE has a timer to track the spent time in the task that you have to set manually.

**3 - Finish the task**
When you finish the task it will merge your branch to the default one, links the commit to the Jira task and set the time spent in Jira task. 

I know many developers that work with time trackers that are manually started/stopped, something automatically (already doing) and integrated with this tools would be a great feature I think.

**Comment by hangxingliu** [comment]
So you hope this extension has features what you mention above in IntelliJ IDEA ?

Maybe I will understand after trying Jira with IDEA.

**Comment by caioflores** [comment]
I'm just thinking about it ahaha. I think that some devs have the problem to track their time while working, because the tools always need this start/stop button way to do it. And this extension make all the coding statistic so transparent that it could help tracking issues time.
I don't know, this kind of automated tracking with tasks/issues was something that I was willing to do in the past months, and I found your repo searching for it.
But maybe this kind of integration should be done in another "Jira extension", and it could integrate with yours to generate the graphs report?
